src/tree.c

```diff
@@ -797,14 +804,35 @@ int git_treebuilder_remove(git_treebuilder *bld, const char *filename)
 
 int git_treebuilder_write(git_oid *oid, git_treebuilder *bld)
 {
+	int error;
+	size_t entrycount;
+	git_buf buffer = GIT_BUF_INIT;
+
+	entrycount = git_strmap_num_entries(bld->map);
+
+	/* Grow the buffer beforehand to an estimated size */
+	error = git_buf_grow(&buffer, entrycount * 72);
+	if (error >= 0)
```

Comment:
This is the opposite of how we do error handling in the library. We use early returns or jumps to cleanup if things go wrong instead of putting the success path inside `if`s.